app/services.go

```diff
@@ -155,15 +157,31 @@ func (s *serviceManager) setupPushService() {
 	if s.err != nil {
 		return
 	}
-	cert, key, err := pushcertificate.Load(
-		s.APNS.CertificatePath,
-		s.APNS.PrivateKeyPass,
-	)
-	if err != nil {
-		s.err = err
-		return
+	var tlscert tls.Certificate
+	var err error
+	if s.APNS.PrivateKeyPath == "" {
+		// note that buford does validity checks where
+		// our direct certificate parsing does not
+		cert, key, err := pushcertificate.Load(
+			s.APNS.CertificatePath,
+			s.APNS.PrivateKeyPass,
+		)
+		if err != nil {
+			s.err = err
+			return
+		}
+		tlscert = pushcertificate.TLS(cert, key)
+	} else {
+		tlscert, err = tls.LoadX509KeyPair(
```

Comment:
If you decide to move the certificate unmarshalling code from the enroll package here, you can just use `pushcertificate.TLS(...)` regardless of wether it's a PEM or a p12.